src/main/java/com/notably/logic/commands/suggestion/SearchSuggestionCommand.java

```diff
@@ -0,0 +1,113 @@
+package com.notably.logic.commands.suggestion;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Queue;
+import java.util.stream.Collectors;
+
+import com.notably.commons.path.AbsolutePath;
+import com.notably.logic.commands.OpenCommand;
+import com.notably.model.Model;
+import com.notably.model.block.BlockTree;
+import com.notably.model.block.BlockTreeItem;
+import com.notably.model.suggestion.SuggestionItem;
+import com.notably.model.suggestion.SuggestionItemImpl;
+
+/**
+ * Represents a suggestion command object to search through all the notes based on keywords.
+ */
+public class SearchSuggestionCommand implements SuggestionCommand {
+    private static final String COMMAND_WORD = "search";
+    private static final String RESPONSE_MESSAGE = "Search through all notes based on keywords";
+    private static final String WHITE_SPACE = "        ";
+
+    private String keywords;
+
+    public SearchSuggestionCommand(String keywords) {
+        Objects.requireNonNull(keywords);
+        this.keywords = keywords;
+    }
+
+    @Override
+    public void execute(Model model) {
+        // Nullity check
+        Objects.requireNonNull(model);
+
+        // Set response text
+        model.setResponseText(RESPONSE_MESSAGE);
+
+        // Set suggestions
+        List<AbsolutePath> allPaths = getAllPaths(model);
+        List<SuggestionItem> suggestions = traverseTree(allPaths, model);
+
+        model.setSuggestions(suggestions);
+    }
+
+    /**
+     * Traverses the whole block tree to get the blocks which contains the keywords.
+     * @param allPaths The list of paths to all the blocks in the block tree.
+     * @param model The app's model.
+     * @return List of SuggestionItem with a display text of the block's path and action of opening the block.
+     */
+    private List<SuggestionItem> traverseTree(List<AbsolutePath> allPaths, Model model) {
+        BlockTree blockTree = model.getBlockTree();
+        List<SuggestionItem> suggestions = new ArrayList<>();
+
+        // Traverse through the whole block tree
+        for (AbsolutePath absolutePath : allPaths) {
+            BlockTreeItem blockTreeItem = blockTree.get(absolutePath);
+            String blockBody = blockTreeItem.getBody().getText();
+
+            /* If a blockBody contains the keywords, we create SuggestionItem by
+               counting the keywords' number of occurrences in the blockBody, setting display text, and setting action
+               to open that particular block when the user chooses that suggestion. */
+            String bodyWithRemovedPunctuation = blockBody.replaceAll("\\p{Punct}", "");
+            String bodyLowerCase = bodyWithRemovedPunctuation.toLowerCase();
```

Comment:
I believe you'd also want to make `keywords` lower case?

Reply:
omg yes sorry i missed this!